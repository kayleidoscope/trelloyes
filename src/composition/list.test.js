// make React available
import React from 'react';

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from 'react-dom';

// make the App component available
import List from './list.js';
import renderer from 'react-test-renderer';

// this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<List />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List 
        header="test" 
        cards="test"
        key="list-test"
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });