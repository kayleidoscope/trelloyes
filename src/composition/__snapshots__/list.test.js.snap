// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  cards="test"
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test
    </h2>
  </header>
  <div
    className="List-cards"
  />
</section>
`;
